### ISSUE TITLE
[JIT] Tensor method API behavior discrepancy, Tensor.detach(..)

### ISSUE THREAD

**Issue by xush6528** [pytorch/pytorch#35990]
In Python, `Tensor.detach()` does not copy the tensor storage. In-place value changes update the original tensor.

See doc, https://pytorch.org/docs/stable/autograd.html#torch.Tensor.detach.

```py
b = a.detach()
b -= c
```
a is changed as well. 
More examples in https://github.com/pytorch/pytorch/issues/6990

In JIT, `Tensor.detach()` copies the tensor storage. As the implementation is, 
https://github.com/pytorch/pytorch/blob/a72946dbab9c1b79a9c75df4f353a341e2ddb19c/aten/src/ATen/native/TensorProperties.cpp#L55-L80

**Comment by zdevito** [comment]
I tried your example in jit/no jit and I am getting the same result in both, which is that the tensor is modified in place as expected. Can you explain more about what your expected a behavior is and what you are actually seeing?

```
import torch

def foo(x):
    x.detach()
    x -= 3

foo_s = torch.jit.script(foo)

a = torch.zeros(3, 4)
foo(a)

b = torch.zeros(3, 4)
foo_s(b)
assert torch.allclose(a, b)
```
import torch

def foo(x):
    x.detach()
    x -= 3

foo_s = torch.jit.script(foo)

a = torch.zeros(3, 4)
foo(a)

b = torch.zeros(3, 4)
foo_s(b)
assert torch.allclose(a, b)
```

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
